### ISSUE TITLE
Parsing stack traces with auto-generated code and delegates

### ISSUE THREAD

**Issue by ThomasArdal** [atifaziz/StackTraceParser#10]
Maybe more of a question than an issue. Do you think it would ever be possible to parse something like this:

```
Microsoft.AspNetCore.Mvc.Controllers.ControllerBinderDelegateProvider.<>c__DisplayClass0_0.<<CreateBinderDelegate>g__Bind|0>d.MoveNext()
```

When parsing this with StackTraceParser the following type and method is resolved:

Type: `Microsoft.AspNetCore.Mvc.Controllers.ControllerBinderDelegateProvider.<>c__DisplayClass0_0.<<CreateBinderDelegate>g__Bind|0>d`
Method: `MoveNext`

The method looks right but type is a bit funny. The outer class here is `ControllerBinderDelegateProvider`. I guess `<>c__DisplayClass0_0` is an auto-generated class from the use of a delegate. And `g__Bind` a method maybe (I lost track there). Not sure what `|0>d` represents.

I think it would be hard to parse this any different, but maybe `<>` or the use of `<` and `>` in a type could be used to better understand the frame. What do you think?

**Comment by atifaziz** [comment]
I answered [a similar question you posted on StackOverflow](https://stackoverflow.com/a/68984918/6682). The parsing is happening correctly. `MoveNext` is indeed the method name and `Microsoft.AspNetCore.Mvc.Controllers.ControllerBinderDelegateProvider.<>c__DisplayClass0_0.<<CreateBinderDelegate>g__Bind|0>d` the type name (including the namespace).
